iceoryx_dds/test/mocks/chunk_mock_dds.hpp

```diff
@@ -49,8 +50,11 @@ class ChunkMockDDS
 
         memset(m_rawMemory, 0xFF, Size);
 
-        m_chunkHeader = new (m_rawMemory) iox::mepoo::ChunkHeader(
-            Size, sizeof(T), alignof(T), iox::CHUNK_NO_CUSTOM_HEADER_SIZE, iox::CHUNK_NO_CUSTOM_HEADER_ALIGNMENT);
+        auto chunkSettingsResult = iox::mepoo::ChunkSettings::create(sizeof(T), alignof(T));
+        iox::cxx::Ensures(!chunkSettingsResult.has_error() && "Invalid parameter for ChunkMock");
+        auto& chunkSettings = chunkSettingsResult.value();
+
+        m_chunkHeader = new (m_rawMemory) iox::mepoo::ChunkHeader(Size, chunkSettings);
```

Comment:
Is m_rawMemory large enough to construct the header in (also considering alignment)?

Reply:
yes, it's large enough and also aligned. See lines 96 and 97. This mock doesn't use the custom header, therefore the payload is always adjacent to the `ChunkHeader`.

But I messed up the `ChunkHeader` tests. I need to provide some memory where the `ChunkHeader` is constructed, since it will write outside of it's bounds if the back-offset is necessary, e.g. when the payload alignment is larger than the `ChunkHeader` alignment. Will fix that.

Reply:
Forget it. I didn't mess up and did the right thing. Just forgot that it's not in the fixture but in `createChunksOnMultipleAddresses`